Source/Scene/Scene.js

```diff
@@ -3645,6 +3664,85 @@ define([
         camera.right = right;
     }
 
+    function updateAsyncLoader(scene, asyncLoader) {
+        var context = scene._context;
+        var uniformState = context.uniformState;
+        var frameState = scene._frameState;
+
+        var view = scene._pickOffscreenView;
+        scene._view = view;
+
+        var ray = asyncLoader.ray;
+        var primitives = asyncLoader.primitives;
+
+        updateCameraFromRay(ray, view.camera);
+
+        updateFrameState(scene);
+        frameState.passes.offscreen = true;
+        frameState.passes.async = true;
+
+        uniformState.update(frameState);
+
+        var commandList = frameState.commandList;
+        var commandsLength = commandList.length;
+
+        var ready = true;
+        var primitivesLength = primitives.length;
+        for (var i = 0; i < primitivesLength; ++i) {
+            var primitive = primitives[i];
+            if (scene.primitives.contains(primitive) && primitive.show) {
+                // Only update primitives that are still contained in the scene's primitive collection and are still visible
+                ready = primitive.update(frameState) && ready;
```

Comment:
Same comment throughout.

Reply:
Also super minor, but
```javascript
ready = primitive.update(frameState) && ready;
```
is better written as
```javascript
ready = (ready && primitive.update(frameState));
```
because
1. It is easier to digest the two `ready`'s in a row
2. Given short circuit expressions, `update` may be avoided - unless this is an intension side effect that you want
3. This avoids the likely L1 cache miss because `update` would likely wipe it and then `ready` would need to be fetched...you will never notice the difference here.

Reply:
This is intentional. `update` needs to be called continually, otherwise tiles that are needed later might be removed from the tileset cache. I'll add a comment to clarify why the order is this way.